src/Microsoft.Performance.Toolkit.Engine/Engine.cs

```diff
@@ -1008,14 +1024,22 @@ private RuntimeExecutionResults ProcessCore()
                          *         data source being processed in more than 1 group
                          * 4. Add a new executor for each IDataSourceGroups picked out in the above step
                          */
+
+                        var dsg = new DataSourceGroup(dataSources, new DefaultProcessingMode());
+                        var processorOptions = optionsResolver.GetProcessorOptions(dsg, processingSource.Instance);
+                        if (!processorOptions.Options.All(o => processingSource.CommandLineOptions.Any(clo => clo.Id.Equals((o.Id as Option)?.Id))))
+                        {
+                            throw new NotSupportedException($"All ProcessorOptions are not supported.");
```

Comment:
Couple of comments on this block:
1. `NotSupportedException` is probably not the most appropriate exception to throw here. `InvalidOperationException` would probably be closer. But...
2. Should we make a custom exception here?
3. "All ProcessorOptions are not supported" is a little vague. It would be more helpful to know exactly what the invalid option is, as well as what processing source is being used

Reply:
Hmmm, I agree with 2, for making a custom exception and I'll make it so the invalid options can be returned.